### ISSUE TITLE
[request] passive mode transmission

### ISSUE THREAD

**Issue by atesin** [loboris/ESP8266_AT_LoBo#5]
hi... thank you for your firmware, it has many new useful features not present in official espressif one =D

however one of the features i really need in my project is passive mode transmission because the tiny size of arduino uart serial buffers often result in data loss ... so i installed your firmware, it worked flawlessly but soon i noticed it lacks this feature :(

so i tried to install official espressif ATv1.7 in my esp-01s 1mb (512+512) flash, but also noticed it doesn't support less than 2mb (1024+1024) by default, so i tried to recompile for 1mb 512+512, but it also lacks documentation/support :(

q1: is there some way to achieve passive trasmission in lobo firmware?
q2: excuse me but if not, could you please give me directions to properly complile official ATv1.7 for 1mb (512+512) flash?
q3: have you plans to support this feature in your firmware in the future?

thank you so much for everything

**Comment by atesin** [comment]
MY BAD

this firmware still has AT+CIPxxx commands (CIPSERVER, CIPSEND, CIPCLOSE) to support passive mode  (additionally has new AT+TCPxxx commands to support receive confirmations instead)

example below, ">" means a serial console typed command, some messages were stripped to improve readability

please mention this in the docs and close this ticket, thank you again for your terrific work with this firmware :D

```
> AT+GMR
AT version:1.7.0.0
ESP8266_AT_LoBo v1.3.1

> AT+CIFSR
+CIFSR:STAIP,"192.168.xx.xx"

> AT+CIPMUX=1
> AT+CIPRECVMODE=1
> AT+CIPSERVER=1,80

# web browser opened at 192.168.x.x
+IPD,0,463

> AT+CIPRECVDATA=0,400
+CIPRECVDATA,400:GET / HTTP/1.1
Host: 192.168.x.x
Connection: keep-alive
Cache-Control: max-age=0
Upgrade-Insecure-Requests: 1
User-Agent: Mozilla/5.0 (Windows NT 6.1; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/78.0.3904.97 Safari/537.36 OPR/65.0.3467.48
Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3
Acce

> AT+CIPRECVLEN?
+CIPRECVLEN:63,0,0,0,0

> AT+CIPRECVDATA=0,63
+CIPRECVDATA,63:pt-Encoding: gzip, deflate
Accept-Language: es-ES,es;q=0.9

> AT+CIPRECVLEN?
+CIPRECVLEN:0,0,0,0,0
```

**Comment by ceeeru** [comment]
Hello Atesin!
How could You open terminal?
I tried it with putty, but only for read, I can't send any command for the device.
I have this:
https://github.com/IOT-MCU/ESP-12S-A9-A9G-GPRS-Node-v1.0/wiki
I can connect to it with putty/serial but can't write anything to terminal window:
![2020-03-20 16_55_21-COM6 - PuTTY]
(https://user-images.githubusercontent.com/58506171/77181478-abb6a300-6acb-11ea-81ce-74baeb44a980.png)
So, what is the serial port settings, to communicate? I can't find any info about it.

Sincerelly:
Norbert

**Comment by atesin** [comment]
> Hello Atesin!
> How could You open terminal?
> I tried it with putty, but only for read, I can't send any command for the device.
> I have this:
> https://github.com/IOT-MCU/ESP-12S-A9-A9G-GPRS-Node-v1.0/wiki
> I can connect to it with putty/serial but can't write anything to terminal window:
> ![2020-03-20 16_55_21-COM6 - PuTTY]
> (https://user-images.githubusercontent.com/58506171/77181478-abb6a300-6acb-11ea-81ce-74baeb44a980.png)
> So, what is the serial port settings, to communicate? I can't find any info about it.
> 
> Sincerelly:
> Norbert

hi norbert ... i guess ctrl+t is to send the char sequence in putty (search in options)

about the serial speed... there is no way to guess it but commonly is 9600 baud rate, you have to try... the config are commonly 8N1 (8 data bits, no parity, 1 stop bit), and NO FLOW CONTROL

i recommend you to use another terminal console application just for serial port instead putty, like coolterm, termite, teraterm, or the serial monitor from arduino ide ... search for it and always explore program options

**Comment by uzi18** [comment]
@ceeeru "ready" means at-firmware is ready, you need to turn on local echo
but indeed please use different terminal app, like hterm, realterm etc.

**Comment by ceeeru** [comment]
It is working!
Thanks for Your suggestion, CoolTerm works on first try!
![2020-03-20 22_14_16-Window](https://user-images.githubusercontent.com/58506171/77206947-7d02f180-6af8-11ea-8df9-f5648ea4bb67.png)

**Comment by ceeeru** [comment]
If I can have another question:
should this board able to connect to internet with simcard? With softwareserial in Arduino/C++ or PPPoS in Micropython (as I saw ESP32 can use PPPoS, but those 2 boards almost the same)?
I'm new in this IOT world, my benefit would be build a GPRS-tracker with communication with my own MQTT server over SSL with x509 client cert.
Am I wait too much from this little board, and should turn to raspberry side to build my own car-tracker solution?
I did my vote with this small footprint panel, but if the hardware is not enough, then I give up.
Micropyhton vould be the best way for me, the C/C++ road is not fully my way.
Thanks for Your answer, have a nice day!
N.
